CloudM.Exercise.WebApi/Controllers/PackageController.cs

```diff
@@ -27,12 +31,63 @@ public async Task<PackageDependencies> Get(string name, string version)
                 PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
             });
 
+            var dependencies = npmPackage.Versions[version].Dependencies;
+
+            foreach (var dep in dependencies)
+            {
+                var subDep = GetDependencies(dep.Key, dep.Value).Result;
+                dependencyTree.Add(dep.Key, new
+                {
+                    Version = dep.Value,
+                    Dependecies = subDep
+                });
+            }
+
             return new PackageDependencies
             {
                 Name = name,
                 Version = version,
-                Dependencies = npmPackage.Versions[version].Dependencies,
+                Dependencies = dependencyTree,
             };
         }
+
+        public async Task<Dictionary<string, dynamic>> GetDependencies(string name, string version)
```

Comment:
This whole method here violates the SRP, because it is a business logic and it has to be moved to a separate service layer, not placed here in the controller